keras/layers/recurrent.py

```diff
@@ -65,7 +65,7 @@ def get_output(self, train=False):
         mask = self.get_output_mask(train)
         if mask:
             # apply mask
-            X *= K.expand_dims(mask)
+            X *= K.expand_dims(mask).astype(X.dtype)
```

Comment:
Please use `K.cast()` instead, I don't think the above would work with the TensorFlow backend.